OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/toolbar/SpellCheck.java

```diff
@@ -0,0 +1,49 @@
+/* 
+ * Copyright (c) 2017 White Hsu
+ * 
+ * See the file LICENSE for copying permission.
+ */
+package latexstudio.editor.toolbar;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import javax.swing.text.BadLocationException;
+import latexstudio.editor.EditorTopComponent;
+import org.openide.awt.ActionID;
+import org.openide.awt.ActionReference;
+import org.openide.awt.ActionReferences;
+import org.openide.awt.ActionRegistration;
+import org.openide.util.Exceptions;
+import org.openide.util.NbBundle.Messages;
+import org.openide.windows.TopComponent;
+import org.openide.windows.WindowManager;
+
+@ActionID(
+        category = "Edit",
+        id = "latexstudio.editor.toolbar.SpellCheck"
+)
+@ActionRegistration(
+        iconBase = "openlatexstudio/icons/spellcheck.png",
+        displayName = "#CTL_SpellCheck"
+)
+@ActionReferences({
+    @ActionReference(path = "Menu/Edit", position = 250),
+    @ActionReference(path = "Toolbars/Comment", position = 3433)
+})
+@Messages("CTL_SpellCheck=Spell Check")
+public final class SpellCheck implements ActionListener {
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        TopComponent tc = WindowManager.getDefault().findTopComponent("EditorTopComponent");
+        EditorTopComponent etc = (EditorTopComponent) tc;        
```

Comment:
We have got our own factory to retrieve the top components - let's use it. The class name is TopComponentFactory and it's used in most of our classes.